### PR TITLE
[CUB-2739] Extract middleware development config/tools with additional typescript config

### PR DESCRIPTION
**Pull request by ignatiusreza**
Most of the code are taken from laboperator-middleware with some changes..

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,60 @@
-# laboperator-middleware-development
\ No newline at end of file
+# laboperator-middleware-development
```
Comment (Jubke) [inline README.md:1]: Should we link to the laboperator-middleware repo here?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
+{
+  "name": "laboperator-middleware-development",
+  "version": "1.0.0",
+  "description": "Laboperator Middleware Development Config & Tools",
+  "repository": "git@github.com:labforward/laboperator-middleware-development.git",
+  "author": "labforward GmbH",
```
Comment (Jubke) [inline package.json:6]:
```suggestion
  "author": "Labforward GmbH",
```

File: src/cli/build.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const fs = require('fs');
+const path = require('path');
+const { execSync } = require('child_process');
+
+module.exports = (argv) => {
+  const dockerfile = path.resolve(__dirname, '../../Dockerfile');
+  const [head, tail = 'develop'] = argv.tag.split(':');
+
+  if (!fs.existsSync('./build')) fs.mkdirSync('./build');
```
Comment (Jubke) [inline src/cli/build.js:9]: Not really a concern right now, but with explicit paths here this will not work on Windows, right?

**Review by Jubke — CHANGES_REQUESTED** [review]
Let's add global settings by adding `.github/settings.yml`:

```YAML
_extends: repo-settings
```

for further reference see:

- https://github.com/labforward/repo-settings
- https://github.com/labforward/driver-ci/blob/develop/.github/settings.yml
- https://github.com/probot/settings
- https://probot.github.io/docs/best-practices/#configuration

File: package.json (hunk 2 above)
Comment (ignatiusreza) [reply package.json:6]: I thought the company legal name is with lowercase `l`?

File: src/cli/build.js (hunk 3 above)
Comment (ignatiusreza) [reply src/cli/build.js:9]: why wouldn't it? wouldn't the node runtime normalize the path?

**Comment by ignatiusreza** [comment]
> for further reference see:
> - labforward/repo-settings

I don't have access to this repo.. :scream:

**Comment by Jubke** [comment]
> I don't have access to this repo.. 

✅  Please try again

**Review by Jubke — APPROVED** [review]
LGTM

File: src/cli/build.js (hunk 3 above)
Comment (Jubke) [reply src/cli/build.js:9]: Right, I meant to comment on the `docker save -o build/...` below, then thought about this and somehow left the comment here... And I think there were other `exec` calls too that included paths.

File: package.json (hunk 2 above)
Comment (Jubke) [reply package.json:6]: Nope

File: src/cli/build.js (hunk 3 above)
Comment (ignatiusreza) [reply src/cli/build.js:9]: right.. those could potentially fails.. not sure.. let's worry about them later when needed.. :see_no_evil:

**Pull request merged**